src/lib.rs

```diff
@@ -190,313 +182,415 @@ pub struct AstarteDeviceDataEvent {
+        interface_path: &str,
+        data: D,
+    ) -> Result<(), Error>
+    where
+        D: TryInto<AstarteType> + Send;
 
-        let mut device = AstarteDeviceSdk {
-            shared: Arc::new(SharedDevice {
-                realm: opts.realm,
-                device_id: opts.device_id,
-                client,
-                events_channel: tx_events,
-                eventloop: Mutex::new(eventloop),
-                interfaces: RwLock::new(opts.interfaces),
-                store: StoreWrapper::new(opts.store),
-            }),
-        };
+    /// Poll updates from the connection implementation, can be placed in a loop to receive data.
+    ///
+    /// This is a blocking function. It should be placed on a dedicated thread/task or as the main
+    /// thread.
+    ///
+    /// ```no_run
+    /// use astarte_device_sdk::{AstarteDeviceSdk, Device, builder::{DeviceBuilder, MqttConfig}};
+    ///
+    /// #[tokio::main]
+    /// async fn main() {
+    ///     let mqtt_config = MqttConfig::new("_", "_", "_", "_");
+    ///
+    ///     let (mut device, mut rx_events) = DeviceBuilder::new()
+    ///         .connect_mqtt(mqtt_config).await.unwrap();
+    ///
+    ///     tokio::spawn(async move {
+    ///         while let Some(event) = rx_events.recv().await {
+    ///             assert!(event.is_ok());
+    ///         };
+    ///     });
+    ///
+    ///     device.handle_events().await;
+    /// }
+    /// ```
+    async fn handle_events(&mut self) -> Result<(), crate::Error>;
+
+    /// Unset a device property.
+    ///
+    /// ```no_run
+    /// use astarte_device_sdk::{AstarteDeviceSdk, Device, builder::{DeviceBuilder, MqttConfig}};
+    ///
+    /// #[tokio::main]
+    /// async fn main() {
+    ///     let mqtt_config = MqttConfig::new("_", "_", "_", "_");
+    ///
+    ///     let (mut device, _rx_events) = DeviceBuilder::new()
+    ///         .connect_mqtt(mqtt_config).await.unwrap();
+    ///
+    ///     device
+    ///         .unset("my.interface.name", "/endpoint/path",)
+    ///         .await
+    ///         .unwrap();
+    /// }
+    /// ```
+    async fn unset(&self, interface_name: &str, interface_path: &str) -> Result<(), Error>;
+}
+
+#[async_trait]
+pub trait InterfaceRegistry {
+    /// Add a new [`Interface`] to the device interfaces.
+    async fn add_interface(&self, interface: Interface) -> Result<(), Error>;
+
+    /// Add a new interface from the provided file.
+    async fn add_interface_from_file(&self, file_path: &str) -> Result<(), Error>;
+
+    /// Add a new interface from a string. The string should contain a valid json formatted
+    /// interface.
+    async fn add_interface_from_str(&self, json_str: &str) -> Result<(), Error>;
+
+    /// Remove the interface with the name specified as argument.
+    async fn remove_interface(&self, interface_name: &str) -> Result<(), Error>;
+}
 
-        device.wait_for_connack().await?;
+#[async_trait]
+pub trait PropertyRegistry {
```

Comment:
This trait should have been superseeded by `PropertyAccess` that was merged in main